keps/sig-node/5526-pod-level-resource-managers/README.md

```diff
@@ -0,0 +1,1856 @@
+check on each container to determine which are eligible for exclusive resource
+allocation. To aid debugging, the results of this check will be surfaced to via
+log message.
+
+New cases that need to be covered:
+
+-   **Pod-level resources set, Container-level resources unset:**
+    -   The alignment decision is made for the pod as a single unit, based on
+        `pod.spec.resources`. The resulting NUMA-aligned resource pool is then
+        managed for the pod, and this entire pool is treated as the shared pool
+        for all containers within it.
+-   **Pod-level resources set, Container-level resources set:**
+    -   The alignment decision is made for the pod as a single unit, based on
+        `pod.spec.resources`. The resulting NUMA-aligned resource pool is then
+        partitioned. Containers that individually meet `Guaranteed` criteria are
+        allocated exclusive slices from the pool, while all other containers are
+        allocated the remaining shared portion of the pool.
+
+Already existing logic for standard and restartable init containers is
+compatible with the changes presented in this KEP. They will continue be
+allocated exclusive slices from the pod's managed resource pool. The resources
+from standard init containers will be made reusable by the app containers within
+the pod's shared pool.
+
+Pod Scope                                                             | Spec                                                                               | Hint Generation | Allocation
+:-------------------------------------------------------------------- | :--------------------------------------------------------------------------------- | :-------------- | :---------
+Current behavior                                                      | Container 1: 3 CPU <br> Container 2: 1 CPU <br> Container 3: 1 CPU                 | {NUMA node 0}   | Container 1: {1, 2, 3} <br> Container 2: {4} <br> Container 3: {5}
+New behavior <br> Pod Level Resources <br> All Guaranteed containers  | Pod: 5 CPU <br> Container 1: 3 CPU <br> Container 2: 1 CPU <br> Container 3: 1 CPU | {NUMA node 0}   | Container 1: {1, 2, 3} <br> Container 2: {4} <br> Container 3: {5}
+New behavior <br> Pod Level Resources <br> Some Guaranteed containers | Pod: 5 CPU <br> Container 1: 3 CPU <br> Container 2 <br> Container 3               | {NUMA node 0}   | Container 1: {1, 2, 3} <br> Container 2: {4, 5} <br> Container 3: {4, 5}
+New behavior <br> Pod Level Resources <br> No Guaranteed containers   | Pod: 5 CPU <br> Container 1 <br> Container 2 <br> Container 3                      | {NUMA node 0}   | Container 1: {1,2,3,4,5} <br> Container 2: {1,2,3,4,5} <br> Container 3: {1,2,3,4,5}
+New behavior <br> Pod Level Resources <br> Admission Failure          | Pod: 5 CPU <br> Container 1: 3 CPU <br> Container 2: 2 CPU <br> Container 3        | Admission Error | Pod Rejected
+
+###### Rejected Configurations
```

Comment:
do we want to reject underutilizing of pod exclusive resources when sum of container resources, all of them are exclusive, is less then Pod's